internal/format/command.go

```diff
@@ -30,3 +33,59 @@ func (c command) GetApplication(extraFields ...string) string {
 	}
 	return fmt.Sprintf("%s %s %s", string(c), getApplicationCommand, strings.Join(extraFields, " "))
 }
+
+// MissingChildren detects missing commands/args.
+// Logic taken from github.com/alecthomas/kong/context.go
+func MissingChildren(node *kong.Node) bool {
+	for _, arg := range node.Positional {
+		if arg.Required && !arg.Set {
+			return true
+		}
+	}
+
+	for _, child := range node.Children {
+		if child.Hidden {
+			continue
+		}
+
+		if child.Argument != nil {
+			if !child.Argument.Required {
+				continue
+			}
+		}
+
+		return true
+	}
+
+	return false
+}
+
+// NoFatalExitIfErrorf prints Usage + friendly message on error (and exits).
+func NoFatalExitIfErrorf(err error, args ...interface{}) error {
```

Comment:
This function name is IMO a bit misleading as it's still a fatal exit (exit code 1).

Reply:
true. changing this into `ExitIfErrorf()`